common-libs/common/src/services/DidDocumentService/KeyVault.ts

```diff
@@ -23,4 +23,11 @@ export interface KeyVault {
    * @param payload payload to be signed
    */
   sign(payload: Buffer): Buffer
+
+  /**
+   * Sign the given payload with the primary public key. Same as sign but async.
+   *
+   * @param payload payload to be signed
+   */
+  signAsync?(payload: Buffer): Promise<Buffer>
```

Comment:
I did not get how this helps me. I still need to implement the sync version of this function right. No one uses the `signAsync` as well? 🤔

Reply:
The issue I had was when implementing a KeyVault backed by metamask, the sign method there is async, and adding an optional async method here was the only work around I could find. Other approaches I could use?